lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2438,6 +2442,39 @@ void ModuleEmitter::emitRTLModule(RTLModuleOp module) {
       addName(module.getArgument(port.argNum), name);
   }
 
+  // Get the list of reserved words we need to avoid.  We could prepopulate this
+  // into the used words cache, but it is large and immutable, so we just query
+  // it when needed.
+  auto &reservedWords = getReservedWords();
+  auto moduleName = module.getName();
+  if (reservedWords.count(moduleName)) {
+    // If not available, we need to auto-unique it.
+    SmallVector<char, 16> nameBuffer(moduleName.begin(), moduleName.end());
+    nameBuffer.push_back('_');
+    auto baseSize = nameBuffer.size();
+
+    // Try until we find something that works.
+    while (1) {
+      auto suffix = llvm::utostr(nextGeneratedNameID++);
+      nameBuffer.append(suffix.begin(), suffix.end());
+      moduleName = StringRef(nameBuffer.data(), nameBuffer.size());
+
+      if (!reservedWords.count(moduleName)) {
+        auto insertResult = usedModuleNames.insert(moduleName);
+        if (insertResult.second) {
+          // If the module name is unique, then the insertion is successfull,
+          // then the return bool is true, then exit from the infinite while
+          // loop.
+          break;
+        }
+      }
```

Comment:
It's probably cleaner to inline the `insertResult`. 

Note: there's a built in assumption (that I think is used other places) that incrementing the suffix won't cause a new conflict with `reservedWords`. There's no Verilog reserved word like `foo_16`, so this is a fine assumption. Maybe it's worth mentioning this.

Reply:
Thanks, inlined it and removed the check for conflict with keywords after adding the suffix. Updated the comments with the note.